### PR TITLE
Dynamic persistence xml

### PR DESCRIPTION
**Pull request by voievodin**
### What does this PR do?
Add ability to generate persistence.xml dynamically which is simpler approach for tests
comparing to the current one, it :

- ensures all dependencies are defined
- it is based on a real server url, user, password, so changes made to those values wouldn't require persistence.xml changes
- allows to adjust jpa related configuration in a single place(module)

Along with that `H2TestHelper.inMemoryDefault()` & `H2TestHelper.shutdownDefault` methods are deprecated, suggested to use H2DBTestServer instead, all the classes using these methods were rewritten to use db server.

#### Changelog
Use dynamic persistence.xml generator in tests

#### Release Notes
N/A

#### Docs PR
N/A

This code was partly reviewed in #4192.
This code is extracted from the [factory_migration](https://github.com/eclipse/che/tree/factory_migration) branch and allows to simplify its review/merge.

### CODE REVIEW

**Comment by voievodin** [comment]
Replaced `XMLTree` with template.

File: core/commons/che-core-commons-test/src/test/java/org/eclipse/che/commons/test/db/PersistTestModuleBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * @author Yevhenii Voevodin
+ */
+public class PersistTestModuleBuilderTest {
+
+    @Test
+    public void generatesPersistenceXml() throws Exception {
+        Path path = new PersistTestModuleBuilder().setDriver("org.h2.Driver")
+                                                  .addEntityClass(MyEntity1.class)
+                                                  .addEntityClass(MyEntity2.class)
```
Comment (sleshchenko) [inline core/commons/che-core-commons-test/src/test/java/org/eclipse/che/commons/test/db/PersistTestModuleBuilderTest.java]: I'd replace one `addEntityClass` that consumes class to one that consumes fqn

File: wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/jpa/JpaTckModule.java [hunk 2]
```diff
@@ -12,40 +12,51 @@
...
-
 /**
  * @author Yevhenii Voevodin
  */
 public class JpaTckModule extends TckModule {
 
     @Override
     protected void configure() {
-        install(new JpaPersistModule("main"));
+        H2DBTestServer server = new H2DBTestServer();
```
Comment (sleshchenko) [inline wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/jpa/JpaTckModule.java]: why not `H2DBTestServer.startDefault();`?

File: wsmaster/integration-tests/postgresql-tck/src/test/java/PostgreSqlTckModule.java [hunk 3]
```diff
@@ -8,16 +8,17 @@
  * Contributors:
  *   Codenvy, S.A. - initial API and implementation
  *******************************************************************************/
+
```
Comment (sleshchenko) [inline wsmaster/integration-tests/postgresql-tck/src/test/java/PostgreSqlTckModule.java:11]: remove extra line

**Review by sleshchenko — APPROVED** [review]
Good job :+1:

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**